File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    answer = []
    row_length = len(arr)
    col_length = max([len(row) for row in arr])
    for row in arr:
        answer.append(row.copy() + [0] * (row_length - col_length))
    if (row_length < col_length):
        for _ in range(col_length - row_length):
            answer.append([0] * col_length)
    return answer

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.java
class Solution {
    public int[][] solution(int[][] arr) {
        int rowLen = arr.length;
        int colLen = 0;
        for (int[] row : arr) {
            colLen = Math.max(colLen, row.length);
        }
        int n = Math.max(rowLen, colLen);
        int[][] answer = new int[n][n];
        for (int i = 0; i < rowLen; i++) {
            for (int j = 0; j < arr[i].length; j++) {
                answer[i][j] = arr[i][j];
            }
        }
        return answer;
    }
}